fedimint-build/src/lib.rs

```diff
@@ -1,20 +1,41 @@
 use std::env;
-use std::fs::File;
-use std::process::Command;
+use std::process::{Command, Stdio};
+
+const FAKE_COMMIT_HASH: &str = "01234569afbe457afa1d2683a099c7af48a523c1";
 
 pub fn set_code_version() {
     if env::var_os("CODE_VERSION").is_none() {
-        let git_hash = if File::open("./../../.git/HEAD").is_ok() {
-            let output = Command::new("git")
-                .args(["rev-parse", "HEAD"])
-                .output()
-                .unwrap();
-            String::from_utf8(output.stdout).unwrap()
+        let output = match Command::new("git")
+            .args(["rev-parse", "HEAD"])
+            .stdout(Stdio::piped())
+            .stderr(Stdio::inherit())
+            .output()
+        {
+            Ok(o) => o,
+            Err(e) => match env::var("ALLOW_GIT_FAIL") {
```

Comment:
No, why can't the builder set CODE_VERSION manually instead of setting this variable

Reply:
In CI Nix will actually not use it, because we don't want to rebuild everything from scratch every time, even if nothing changed, just because git hash is different.

But yeah. `FEDIMINT_VERSION_HASH=...` should make the build not run `git ...` at all, and just take and use it. And if it wasn't set that means we *require* `git` to succeed.